### PR TITLE
Compass rose fix

### LINKED ISSUE #6257: Compass rose isn't being transformed properly
#### Summary
For cameras other than the nav cam, toggling the compass rose from FOV "up" to north "up" doesn't properly transform the equipment (spacecraft/rover) body SVG element properly.

Initially loaded FOV "up" for aftcam:
Screen Shot 2023-02-01 at 4 37 16 PM

Having toggled the rose to north "up":
Screen Shot 2023-02-01 at 4 37 21 PM

Note erroneous position of the body element.

#### Expected vs Current Behavior
The equipment body element should be rotated in registration with FOV element.

#### Solution
The FOV indicator and rose ticks (green and yellow arrows) are rotated on toggle which is correct, but the equipment inner `__body` element is also being rotated on toggle - this is wrong. That element should only be transformed for cases where the camera rotates with respect to the body, like a gimbaled camera.
Screen Shot 2023-02-01 at 5 08 09 PM

Instead, all of `cr-vrover` should rotate with the FOV. The fix here therefore (maybe?) is:
1. Move `cr-vrover` into the `c-cr__cam-fov` element, and rename that for clarity. Continue to rotate that element as current.
2. Remove the rotation on `cr-vrover__body` that is firing in response to compass toggling; instead, make sure that element is only transformed in the case of a gimbaled camera.

#### Steps to Reproduce

1. navigate to a navcam image endpoint
2. see that the equipment body is oriented properly
3. see that the camera is oriented properly in relation to the equipment
4. click on the compass rose to lock north up on the rose
5. see that the entire compass rose rotates correctly and equipment and camera are still in relation to each other
6. repeat steps with a hazcam
7. repeat steps with an aftcam

#### Environment

* Open MCT Version:  2.1.6-SNAPSHOT
* Deployment Type:  /testathon
* OS:
* Browser:

#### Impact Check List

- [x] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by davetsay**
Closes #6257

### Describe your changes:
* Fixes equipment detached from camera
* Fixes rotation of compass dial in relation to non-gimballing cameras
* Fixes compass HUD display for non-gimballing cameras

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [x] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Review by ozyx — CHANGES_REQUESTED** [review]
One minor comment, pointing out something that could evaluate to `NaN` and bypass fallback code

**Review by akhenry — CHANGES_REQUESTED** [review]
Looks good to me. Just the issue that @ozyx  highlighted, which I think we need to address.

File: src/plugins/imagery/components/Compass/Compass.vue [hunk 1]
```diff
@@ -62,11 +68,14 @@ export default {
...
+        hasGimble() {
+            return this.cameraAzimuth !== undefined;
         },
-        showCompassRose() {
-            return (this.hasCameraPan || this.hasHeading) && this.cameraAngleOfView > 0;
+        // compass ordinal orientation of camera
+        normalizedCameraAzimuth() {
+            return this.hasGimble
+                ? rotate(this.cameraAzimuth)
+                : rotate(this.heading, -this.transformations?.rotation ?? 0);
```
[the remark this answers is not in the document]
Comment (davetsay) [reply src/plugins/imagery/components/Compass/Compass.vue:78]: thanks!

[the remark this answers is not in the document]
Comment (davetsay) [reply src/plugins/imagery/components/Compass/Compass.vue]: also made the change that transformations will always be defined here.

**Comment by akhenry** [comment]
Found an issue in testing. If I change the bounds while I'm on an image the image reloads but the compass does not.

1. Navigate to an image with bounds set such that at least one image is in bounds (if necessary)
2. Expand the time window
3. Observe that the image reloads but the compass does not.

**Review by charlesh88 — APPROVED** [review]
From my perspective (non-coding) LGTM.

**Review by akhenry — APPROVED** [review]
LGTM

**Review by ozyx — APPROVED** [review]

**Pull request merged**